pkg/test/nodes.go

```diff
@@ -74,6 +74,14 @@ func NodeClaimLinkedNode(nodeClaim *v1beta1.NodeClaim) *v1.Node {
 				Labels:      nodeClaim.Labels,
 				Annotations: nodeClaim.Annotations,
 				Finalizers:  nodeClaim.Finalizers,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: "karpenter.sh/v1beta1",
+						Kind:       "NodeClaim",
+						Name:       nodeClaim.Name,
+						UID:        "6ad01199-df8b-49dc-abde-5673d730f735",
```

Comment:
I've had to deal with too many flaky test cases caused by randomness, so I prefer to set fixed UIDs in any test cases I write.

That said, I'm going to remove this anyway since we don't need to set OwnerRefs for finding NodePools thanks to your other suggestion.